src/Generation/generator.py

```diff
@@ -1,62 +1,63 @@
 import json
-import regex
-from llm.connector import connector
+import random
 import importlib
 
-DESCRIPTION_DIRECTORY = "Generation/Descriptions"
-FORM_FILE_NAME = "form.py"
-QUERY_FILE_NAME = "query_text"
-EXAMPLES_FILE_NAME = "examples.json"
-PREDEFINED_FILE_NAME = "predefined.json"
-FORM_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + FORM_FILE_NAME
-QUERY_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + QUERY_FILE_NAME
-EXAMPLES_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + EXAMPLES_FILE_NAME
-PREDEFINED_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + PREDEFINED_FILE_NAME
-FORM_MODUL_NAME = "form"
-FORM_CLASS_NAME = "form"
-EXAMPLES_KEY = "Examples"
-NUMBER_OF_TRIES_TO_GENERATE = 2
+import llm
 
-class Generator:
-    __json_pattern = regex.compile(r'\{(?:[^{}]|(?R))*\}')
+_DESCRIPTION_PATH = 'src/Generation/Descriptions/{type_name}'
+_MODEL_FILE = _DESCRIPTION_PATH + '/model.py'
+_QUERY_FILE = _DESCRIPTION_PATH + '/query_text'
+_EXAMPLES_FILE = _DESCRIPTION_PATH + '/examples.json'
+_PREDEFINED_FILE = _DESCRIPTION_PATH + '/predefined.json'
+_MODEL_MODULE_NAME = "model"
+_MODEL_CLASS_NAME = "Model"
+
+_generator_cache: dict[str, 'Generator'] = {}
+
+def llm_create(type_name: str, count=1, **kwargs) -> list[dict]:
+    if type_name in _generator_cache:
+        generator = _generator_cache[type_name]
+    else:
+        generator = Generator(type_name)
+        _generator_cache[type_name] = generator
+    return generator.generate(count, **kwargs)[1]
 
+class Generator:
     def __init__(self, type_name: str):
         self._type_name = type_name
-        loader = importlib.machinery.SourceFileLoader(FORM_MODUL_NAME, FORM_FILE.format(type_name=type_name))
+        loader = importlib.machinery.SourceFileLoader(_MODEL_MODULE_NAME, _MODEL_FILE.format(type_name=type_name))
         module = loader.load_module()
-        self._form = getattr(module, FORM_CLASS_NAME)
-        self._template_query_text = open(QUERY_FILE.format(type_name=type_name), 'r').read()
-        examples = json.load(open(EXAMPLES_FILE.format(type_name=type_name), 'r'))
-        self._examples = []
-        for example in examples[EXAMPLES_KEY]:
-            self._examples.append(json.dumps(example, indent=4))
-
-    def generate(self, **kwargs) -> (bool, dict):
-        query_text = self._template_query_text.replace("{{form}}", json.dumps(self._form().model_dump(), ensure_ascii=False, indent=4))
-        for (key, value) in kwargs.items():
+        self._model = getattr(module, _MODEL_CLASS_NAME)
+        with open(_QUERY_FILE.format(type_name=type_name), 'r') as f:
+            self._template_query_text = f.read()
+
+        try:
+            with open(_EXAMPLES_FILE.format(type_name=type_name), 'r') as f:
```

Comment:
I think the examples could be just deleted at this point and they can be simply written into the query_text if necessary.
Probably makes easier to fix the enemy_type too.

Reply:
Having it in separate file is useful, because this way Pydantic also checks that your examples are valid.

If we want to get rid of that file, could we instead move the examples to model.py like this:
```python
EXAMPLES = [
  Model(foo='a', bar='Hello'),
  Model(foo='b', bar='World')
]
```

Reply:
Ohh, I haven't noticed that it is passed to the model. In this case, yeah, you can leave it as it is.